### ISSUE TITLE
Getting -nan as a weight when training a model

### ISSUE THREAD

**Issue by plivesey** [collinhundley/Swift-AI#86]
My model is failing to train a NN. All the weights just get set to -nan so infer always fails (returning -nan). Do you have any advice on how to debug?

Sorry for such a vague issue. I'm only able to reproduce this on my training set (not on a super simple, contrived dataset). Hopefully, you've seen issues like this before and can point me in the right direction to give you more info.

Btw, this is a super awesome library. I am really disappointed that CoreML doesn't allow training a NN. The classification of this library is SO MUCH FASTER. Talking about much less than 1 ms compared to 100 ms.

**Comment by plivesey** [comment]
Just to guess at something which looks a bit weird...all my biases are zero. So when I put a breakpoint at:

`cache.layerBiases[layer] = cache.newLayerBiases[layer]`

newLayerBiases is zero for every entry. This is unexpected right?

**Comment by plivesey** [comment]
I have a guess at what may be going wrong. I am using 1 as a batch size. I am trying to simplify things and so just want to train all my data as one batch. Is this possible? Maybe what I'm actually doing is training one row at a time instead of batching everything together...

**Comment by plivesey** [comment]
Problem solved! This library doesn't work for a single labeled output. I was outputing: [1] as a label for positive and [0] as a label for negative. Instead, you need to do:

[1, 0] for positive and [0, 1] for negative!

Thanks for this awesome library. Finally making progress.
